### PR TITLE
[DPE-5762] Add CI workflows

### PR DESCRIPTION
**Pull request by reneradoi**
# Summary
In order to build, test and publish the etcd operator automatically, we will run Github action workflows when creating or pushing to PR's or merging PR's to the default branch. These workflows will be based on `data-platform-workflows`.

# Note
We are currently still using:
- Ubuntu 22.04 base (due to https://github.com/canonical/data-platform-workflows/issues/246, `data-platform-workflows` don't support 24.04 yet)
- Juju 3.5.4 as default for integration testing (3.6 has not yet been released to stable, but is already added to scheduled test runs)

Furthermore the release job for publishing to Charmhub is not active yet because credentials still need to be added there.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+on:
+  workflow_call:
+  pull_request:
+  schedule:
+    - cron: '53 0 * * *' # Daily at 00:53 UTC
+
+jobs:
+  lint:
+    name: Lint
+    uses: canonical/data-platform-workflows/.github/workflows/lint.yaml@v22.0.0
```
Comment (skourta) [inline .github/workflows/ci.yaml:19]: I think we should update to the latest data platform workflows

File: metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Copyright 2024 Canonical Ltd.
```
Comment (skourta) [inline metadata.yaml:1]: the `metadata.yaml` is no longer needed since charmcraft 2.5 I believe the only file needed now is `charmcraft.yaml`

**Review by skourta — APPROVED** [review]
I left minor comments

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (reneradoi) [reply .github/workflows/ci.yaml]: Agree.

File: metadata.yaml (hunk 2 above)
Comment (reneradoi) [reply metadata.yaml:1]: Unfortunately we have to keep it due to https://github.com/canonical/data-platform-workflows/issues/169. I've started without, but CI job fail then.

**Pull request merged**